### PR TITLE
add retry loop to start command

### PR DESCRIPTION
**Pull request by cblgh**

### CODE REVIEW

File: sim/dsl.go [hunk 1]
```diff
@@ -374,20 +374,42 @@ func (s Simulator) execute() {
...
+			// a retry loop that tries to ping puppet's sbot, exits when ok or max retries reached
+			MAX_RETRIES := 10
+			var feedID string
+			for retries := 0; retries < MAX_RETRIES; retries++ {
+				feedID, err = DoWhoami(p)
+				if err == nil {
+					break
+				} else {
+					canceled := s.checkCancellation()
+					if canceled {
```
Comment (cryptix) [inline sim/dsl.go:388]:
```suggestion
					if s.checkCancellation() {
```

**Review by cryptix — APPROVED** [review]
just a few nitpicks but SGTM

File: sim/dsl.go [hunk 2]
```diff
@@ -374,20 +374,42 @@ func (s Simulator) execute() {
 				instr.TestFailure(err)
 				continue
 			}
-			// TODO: make this more resource efficient (have a retry loop that tries to ping puppet's sbot, exit when ok)
-			// look at waituntil's implementation for a good solution?
 			sleeper.sleep(1 * time.Second)
```
[the remark this answers is not in the document]
Comment (cblgh) [reply sim/dsl.go:377]: i actually did remove it and then noticed its lack was problematic for reasons of not giving the sbot process enough time to start & reinstated it. without an initial sleep, there's an error shown pretty much always when starting a puppet, and i thought that was inutile noise :)

File: sim/dsl.go (hunk 1 above)
Comment (cblgh) [reply sim/dsl.go]: :+1:

**Pull request merged**